src/lib/registration/dialogs/addons_selection.rb

```diff
@@ -0,0 +1,135 @@
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "yast"
+require "cwm/dialog"
+require "registration/ui/abort_confirmation"
+require "registration/controllers/addons_selection"
+require "registration/widgets/master_detail_selector"
+
+module Registration
+  module Dialogs
+    # The dialog to display the addons selection for registered products
+    class AddonsSelection < CWM::Dialog
+      Yast.import "Stage"
+
+      # @return [Registration::Registration] the object handling the communication with SCC
+      attr_reader :registration
+
+      # Constructor
+      #
+      # @param registration [Registration::Registration]
+      def initialize(registration)
+        textdomain "registration"
+
+        @registration = registration
+
+        super()
+      end
+
+      # @macro seeDialog
+      def title
+        _("Extension and Module Selection")
+      end
+
+      # @macro seeDialog
+      def contents
+        VBox(
+          Left(Heading(_("Available Extensions and Modules"))),
+          addons_selector_widget
+        )
+      end
+
+      # @macro seeDialog
+      def run
+        ret = super until continue_actions.include?(ret)
+
+        case ret
+        when :next
+          ret = handle_next_action
+        when :cancel, :abort
+          ret = handle_cancel_action
+        end
+
+        ret
+      end
+
+      # Handles the cancel action
+      #
+      # @return [nil, :abort] nil if user regrets about canceling; :abort otherwise
+      def handle_cancel_action
+        return nil if Yast::Stage.initial && !UI::AbortConfirmation.run
+
+        controller.restore_selection
+        :abort
+      end
+
+      # Handles the next action
+      #
+      # This action will be canceled if there are selected too many addons requiring a registration
+      # code. Ssee {Registration::Controllers::AddonsSelection#supported_addon_count}
+      #
+      # @return [nil, :skip, :abort] nil if there are too many addons requiring reg. code;
+      #                              :skip when there are no selected addons;
+      #                              :next otherwise
+      def handle_next_action
+        return nil unless controller.supported_addon_count?
+
+        controller.selected_items.empty? ? :skip : :next
+      end
+
+      # @return [Array<Symbol>]
+      def continue_actions
+        @continue_actions ||= [:next, :back, :cancel, :abort, :skip]
+      end
+
+      # @macro seeDialog
+      def help
+        [
+          _("<p>Here you can select available extensions and modules for your system.</p>"),
```

Comment:
I tried to keep the translation as untouched as possible, but I has to change some of them a little bit. Shall we simply change them as needed?

Reply:
So, maybe its better to use a "single" translation. Right?

Reply:
Well, it depends. One big translation has a disadvantage that if you change just one word all  translations ares invalidated. With smaller parts only that smaller parts needs to be updated by translators.

And as you already wrote "I has to change some of them a little bit" it seems that having more smaller parts is better here.